spec/src/main/asciidoc/jakarta-ee.adoc

```diff
@@ -1,24 +1,42 @@
+For example, a Jakarta Data provider might use a CDI portable extension to integrate with dependency injection.
+The role of the Jakarta Data provider is to supply implementations of application-defined repositories.
+
+A Jakarta Data provider must supply an implementation of each repository interface which either:
+
+- explicitly specifies the name of the provider via the `provider` member of the `@Repository` annotation, or
+- does not explicitly specify a `provider`, but whose entity classes are annotated with an entity-defining annotation type supported by the provider.
+
+The repository implementation, and the provider itself, is responsible for mediating interaction between the application program and the datastore, often taking advantage of other Jakarta EE services such as datasource provisioning and transaction management.
+
+An application usually obtains a reference to an implementation of a repository via dependency injection.
+Therefore, the provider must make its repository implementations available to the bean container responsible for dependency injection.
+
+Multiple Jakarta Data provider may coexist within a single program, and provide repository implementations.
+It is the responsibility of the application developer to ensure that each repository is eligible for implementation by exactly one provider, according to the rules specified below.
 
 === Provider support for Entities
 
-A Jakarta Data provider supplies an implementation of repository interfaces in Jakarta Data for one or more types of entities. An entity refers to a class that represents objects in a storage engine, such as SQL or NoSQL databases.
+A Jakarta Data provider supports one or more entity-defining annotation types.
+For example:
 
-The `jakarta.persistence.Entity` annotation from the Jakarta Persistence specification can be used by repository entity classes for Jakarta Data providers that are backed by a Jakarta Persistence provider. Other Jakarta Data providers must not support the `jakarta.persistence.Entity` annotation.
+- The `jakarta.persistence.Entity` annotation from the Jakarta Persistence specification is the entity-defining annotation for Jakarta Data providers backed by a Jakarta Persistence provider. Other Jakarta Data providers must not support the use of `jakarta.persistence.Entity` as an entity-defining annotation.
 
-The `jakarta.nosql.Entity` annotation from the Jakarta NoSQL specification can be used by repository entity classes for Jakarta Data providers that are backed by NoSQL databases. Other Jakarta Data providers must not support the `jakarta.nosql.Entity` annotation.
+- The `jakarta.nosql.Entity` annotation from the Jakarta NoSQL specification is the entity-defining annotation for Jakarta Data providers backed by NoSQL databases. Other Jakarta Data providers must not support the use of `jakarta.nosql.Entity` as an entity-defining annotation.
```

Comment:
To allow for providers that support unannotated entities:
```suggestion
- The `jakarta.nosql.Entity` annotation from the Jakarta NoSQL specification is an entity-defining annotation for Jakarta Data providers backed by NoSQL databases. Other Jakarta Data providers must not support the use of `jakarta.nosql.Entity` as an entity-defining annotation.
```

Reply:
Applied the change.